lib/ProductOpener/Ingredients.pm

```diff
@@ -749,7 +771,7 @@ my %min_regexp = (
 
 my %ignore_strings_after_percent = (
 	en => "of (?:the )?(?:total weight|grain is wholegrain rye)",
-	es => "(?:en el chocolate(?: con leche)?)",
+	es => "(?:en el chocolate(?: con leche)?)|(?:de)",
 	fi => "jauhojen määrästä",
 	fr => "(?:dans le chocolat(?: (?:blanc|noir|au lait))?)|(?:du poids total|du poids)",
```

Comment:
```suggestion
	fr => "(?:dans le chocolat(?: (?:blanc|noir|au lait))?)|(?:du poids total|du poids)|(?:de)",
```

Reply:
We can do it differently by using the $of regexp. This works well:

```
# 50% beef, 20g of oranges
				# 90% boeuf, 100% pur jus de fruit, 45% de matière grasses
				my $of = $of{$ingredients_lc} || ' ';    # default to space in order to not match an empty string
				if ($ingredient =~ /^\s*$percent_or_quantity_regexp(?:$of|\s)+/i) {
					$percent_or_quantity_value = $1;
					$percent_or_quantity_unit = $2;
```

Reply:
I'll add a test for it.